orange/src/test/java/com/hartwig/hmftools/orange/algo/linx/LinxBreakendInterpreterTest.java

```diff
@@ -0,0 +1,161 @@
+package com.hartwig.hmftools.orange.algo.linx;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import com.hartwig.hmftools.common.gene.TranscriptRegionType;
+import com.hartwig.hmftools.common.linx.LinxBreakend;
+import com.hartwig.hmftools.common.linx.LinxSvAnnotation;
+import com.hartwig.hmftools.common.linx.LinxTestFactory;
+import com.hartwig.hmftools.common.purple.PurpleTestUtils;
+import com.hartwig.hmftools.common.sv.StructuralVariant;
+import com.hartwig.hmftools.common.sv.StructuralVariantType;
+import com.hartwig.hmftools.datamodel.linx.LinxBreakendType;
+import com.hartwig.hmftools.orange.algo.pave.TestEnsemblDataCacheFactory;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+public class LinxBreakendInterpreterTest
+{
+
+    private static final String GENE = "gene";
+    private static final String VCF_ID = "vcfId";
+    private static final double EPSILON = 0.000001D;
+
+    @Test
+    public void canInterpretValidData()
+    {
+        List<StructuralVariant> structuralVariants = List.of(
+                createStructuralVariant()
+        );
+
+        List<LinxSvAnnotation> linxSvAnnotations = List.of(createSvAnnotation());
+        List<LinxBreakend> breakends = createBreakends();
+
+        LinxBreakendInterpreter interpreter = createInterpreter(structuralVariants, linxSvAnnotations);
+        com.hartwig.hmftools.datamodel.linx.LinxBreakend left = interpreter.interpret(breakends.get(0));
+        assertEquals("7", left.chromosome());
+        assertEquals("q34", left.chromosomeBand());
+        assertEquals(LinxBreakendType.DUP, left.type());
+        assertEquals(-1, left.orientation());
+        assertEquals(1.5D, left.junctionCopyNumber(), EPSILON);
+
+        com.hartwig.hmftools.datamodel.linx.LinxBreakend right = interpreter.interpret(breakends.get(1));
+        assertEquals("7", right.chromosome());
+        assertEquals("q34", right.chromosomeBand());
+        assertEquals(LinxBreakendType.DUP, right.type());
+        assertEquals(1, right.orientation());
+        assertEquals(1.5D, right.junctionCopyNumber(), EPSILON);
+    }
+
+    @Test
+    public void canFallbackWhenNoSvAnnotation()
+    {
+        List<StructuralVariant> structuralVariants = List.of(
+                createStructuralVariant()
+        );
+
+        List<LinxSvAnnotation> linxSvAnnotations = List.of();
+        List<LinxBreakend> breakends = createBreakends();
+
+        LinxBreakendInterpreter interpreter = createInterpreter(structuralVariants, linxSvAnnotations);
+        com.hartwig.hmftools.datamodel.linx.LinxBreakend left = interpreter.interpret(breakends.get(0));
+        assertEquals("", left.chromosome());
+        assertEquals("q34", left.chromosomeBand());
```

Comment:
Haven't looked in detail but seems weird to have a chromosome band without a chromosome? Is this intended behaviour?

Reply:
The chromosome is extracted from the associated structural variant, switching on if the breakend is start or not. Turns out the chromosome band (following the original derivation) just comes from the gene via the ensembl cache. Since the gene is available in the breakend, it's technically possible for us to fill this in without the chrom if the structural variant can not be resolved for some reason (will result in warning in the log). It does look weird.

Reply:
Clear. Don't think its super important to "fix", hopefully the situation is rare anyways :)